### PR TITLE
feat: query level data cache metrics

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- query level data cache metrics
- show cache metrics in "system.query_log"
  three new columns introduced `bytes_from_remote_disk`, `bytes_from_local_disk`, `bytes_from_memory`

Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14313)

### CODE REVIEW

File: src/query/catalog/src/statistics/data_cache_statistics.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// limitations under the License.
+
+use std::sync::atomic::AtomicUsize;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Default)]
+pub struct DataCacheMetrics {
+    bytes_from_storage: AtomicUsize,
```
Comment (BohuTANG) [inline src/query/catalog/src/statistics/data_cache_statistics.rs:22]: `bytes_from_remote_storage`?

File: src/query/catalog/src/statistics/data_cache_statistics.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    bytes_from_disk_cache: AtomicUsize,
```
Comment (BohuTANG) [inline src/query/catalog/src/statistics/data_cache_statistics.rs:23]: `bytes_from_local_disk`?

File: src/query/catalog/src/statistics/data_cache_statistics.rs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    bytes_from_mem_cache: AtomicUsize,
+}
+
+#[derive(Default, Clone, Serialize, Deserialize)]
+pub struct DataCacheMetricValues {
+    pub bytes_from_storage: usize,
+    pub bytes_from_disk_cache: usize,
+    pub bytes_from_mem_cache: usize,
+}
```
Comment (BohuTANG) [inline src/query/catalog/src/statistics/data_cache_statistics.rs:32]:
```suggestion
pub struct DataCacheMetricValues {
    pub bytes_from_remote_disk: usize,
    pub bytes_from_local_disk: usize,
    pub bytes_from_memory: usize,
}
```

File: src/query/catalog/src/statistics/data_cache_statistics.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::atomic::AtomicUsize;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Default)]
+pub struct DataCacheMetrics {
```
Comment (BohuTANG) [inline src/query/catalog/src/statistics/data_cache_statistics.rs:21]: Make these fileds name also same as struct of `DataCacheMetricValues`?

Comment (dantengsky) [reply src/query/catalog/src/statistics/data_cache_statistics.rs:21]: :ok_hand:

**Review by BohuTANG — APPROVED** [review]

File: src/query/service/src/api/rpc/exchange/statistics_receiver.rs [hunk 5]
```diff
@@ -142,6 +142,11 @@ impl StatisticsReceiver {
                 ctx.get_merge_status().write().merge_status(status);
                 Ok(false)
             }
+            Ok(Some(DataPacket::DataCacheMetrics(metrics))) => {
+                log::info!("merge query data cache metrics");
```
Comment (BohuTANG) [inline src/query/service/src/api/rpc/exchange/statistics_receiver.rs:146]: This log is unused?

**Pull request merged**